deploy-agent/deployd/agent.py

```diff
@@ -87,10 +88,22 @@ def load_status_file(self):
         self._config.update_variables(self._curr_report)
 
     def first_run(self):
-        """ check if this the very first run of agent on this instance """
+        """ check if this the very first run of agent on this instance
+            supports sticky Truth: re-evaluate first run
+            only if self._first_run is None or False
+            return: bool self._first_run
+        """
+        if self._first_run:
```

Comment:
i think you can set the `__init__` value to false and collapse these 2 if cases into 
```
if self._first_run or not self._envs:
    self._first_run = True
```
and get rid of the else

Reply:
This also gets rid of the possibility of the `None` value which is confusing since it's never an evaluated state.

Reply:
Both great suggestions.  Added.  Thank you!